### PR TITLE
Sort at query time

### PR DESCRIPTION
**Pull request by Kerollmops**
Re-introduce the Sort at the query time (https://github.com/meilisearch/milli/issues/305)

### CODE REVIEW

**Comment by curquiza** [comment]
⚠️ Should wait for the release of v0.21.0 before any merge

**Review by curquiza — APPROVED** [review]
I approve the revert of revert

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**
